src/features/counterfactual/CounterfactualSuccessScreen.tsx

```diff
@@ -18,7 +18,7 @@ const CounterfactualSuccessScreen = () => {
   const [networks, setNetworks] = useState<ChainInfo[]>([])
   const addressBooks = useAllAddressBooks()
   const safeName = safeAddress && chain ? addressBooks?.[chain.chainId]?.[safeAddress] : ''
-  const isCFCreation = !!networks.length
+  const isCFCreation = !!networks.length && !chainId
```

Comment:
I had to do some digging to understand why this works. It feels a bit disconnected / hard to follow.

Could we not store a state with the `SafeCreationEvent` instead and set it to `SafeCreationEvent.INDEXED` if we receive an indexed event and to `SafeCreationEvent.AWAITING_EXECUTION` if we receive that event?

Then we could change this quite easily to 
```suggestion
  const isCFCreation = event === SafeCreationEvent.AWAITING_EXECUTION
```